core/src/main/java/io/undertow/server/protocol/ajp/AjpOpenListener.java

```diff
@@ -55,6 +55,8 @@
  */
 public class AjpOpenListener implements OpenListener {
 
+    private static final String DEFAULT_AJP_ALLOWED_REQUEST_ATTRIBUTES_PATTERN = System.getProperty("io.undertow.ajp.allowedRequestAttributesPattern");
```

Comment:
Should this use a privileged action?

Reply:
@jamezp in fact, I have missed that fact when reviewing this earlier!
We do have a SecurityActions class for this, I'll update the PR since @msfm is in another time zone

Reply:
Thanks for pointing this out. I've just found that privileged action is not used when executing System.getPropery() in other places. So, I think we will also need to update others later.